File: BungiiAutomation/Bungii.Test.Regression.Android.Integration/Pages/LoginSignupPages/SignupPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Bungii.Test.Regression.Android.Integration.Pages
{
    class SignupPage
    {
        public SignupPage(IWebDriver driver)
        {
           PageFactory.InitElements(driver, this);
        }

        // Login link
        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/title_login")]
        public IWebElement Link_Login { get; set; }

        //-------------Signup fields---------------------------------------------------------------------------
        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_field_first_nam")]
        public IWebElement TextField_FirstName { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_field_last_name")]
        public IWebElement TextField_LastName { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_field_email")]
        public IWebElement TextField_Email { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_field_phone")]
        public IWebElement TextField_Phonenumber { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_field_password")]
        public IWebElement TextField_Password { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_field_prmo_code")]
        public IWebElement TextField_Referral { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_chevron_right")]
        public IWebElement Select_ReferralSource { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.CheckedTextView[@text='Other']")]
        public IWebElement Option_ReferralSource { get; set; }

        [FindsBy(How = How.Id, Using = "android:id/button1")]
        public IWebElement Link_ReferralSourceDone { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_textview_hear_about_us_choice")]
        public IWebElement Text_ReferralSource { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signupGlobalButton")]
        public IWebElement Button_Signup { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/buttonPanel")]
        public IWebElement Popup_ReferralCode { get; set; }

        [FindsBy(How = How.Id, Using = "android:id/button2")]
        public IWebElement Button_NoReferralConfirm { get; set; }

        [FindsBy(How = How.Id, Using = "android:id/button1")]
        public IWebElement Button_NoReferralYes { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/snackbar_text")]
        public IWebElement Snackbar_Error { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/signup_textview_hear_about_us_choice")]
        public IWebElement Text_ReferralSourceAdded { get; set; }

        //--------------Sign up fields error messages-----------------------------------------------------------
        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@resource-id='com.bungii.customer:id/textinput_error' and @instance='2']")]
        public IWebElement Cust_Signup_Error_Email { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@resource-id='com.bungii.customer:id/textinput_error' and @instance='3']")]
        public IWebElement Cust_Signup_Error_Phone { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@resource-id='com.bungii.customer:id/textinput_error' and @instance='4']")]
        public IWebElement Cust_Signup_Error_Password { get; set; }

        //--------------Verification page elements---------------------------------------------------------------

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/field_sms_code")]
        public IWebElement Textfield_SMSCode { get; set; }

        [FindsBy(How = How.Id, Using = "com.bungii.customer:id/smsVerifyContinue")]
        public IWebElement Button_VerifyContinue { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.Button[@text='RESEND']")]
        public IWebElement Link_Resend { get; set; }

        [FindsBy(How = How.XPath, Using = "//android.widget.TextView[@text='VERIFICATION']")]
        public IWebElement Title_Verification { get; set; }
    }
}

File: Bungii.Test.Integration/src/test/java/com/bungii/android/pages/bungiiCustomer/SignupPage.java
package com.bungii.android.pages.bungiiCustomer;

import com.bungii.common.core.PageBase;
import org.openqa.selenium.WebElement;

public class SignupPage extends PageBase {

    // Login link
    public WebElement Link_Login() { return findElement("com.bungii.customer:id/title_login", LocatorType.Id); }
    //-------------Signup fields---------------------------------------------------------------------------
    public WebElement TextField_FirstName() { return findElement("com.bungii.customer:id/signup_field_first_name", LocatorType.Id); }

    public WebElement TextField_LastName() { return findElement("com.bungii.customer:id/signup_field_last_name", LocatorType.Id); }

    public WebElement TextField_Email() { return findElement("com.bungii.customer:id/signup_field_email", LocatorType.Id); }

    public WebElement TextField_Phonenumber() { return findElement("com.bungii.customer:id/signup_field_phone", LocatorType.Id); }

    public WebElement TextField_Password() { return findElement("com.bungii.customer:id/signup_field_password", LocatorType.Id); }

    public WebElement TextField_Referral() { return findElement("com.bungii.customer:id/signup_field_prmo_code", LocatorType.Id); }

    public WebElement Select_ReferralSource() { return findElement("com.bungii.customer:id/signup_chevron_right", LocatorType.Id); }

    public WebElement Option_ReferralSource() { return findElement("//android.widget.CheckedTextView[@text='Other']", LocatorType.XPath); }

    public WebElement Link_ReferralSourceDone() { return findElement("android:id/button1", LocatorType.Id); }

    public WebElement Text_ReferralSource() { return findElement("com.bungii.customer:id/signup_textview_hear_about_us_choice", LocatorType.Id); }

    public WebElement Button_Signup() { return findElement("com.bungii.customer:id/signupGlobalButton", LocatorType.Id); }

    public WebElement Popup_ReferralCode() { return findElement("com.bungii.customer:id/buttonPanel", LocatorType.Id); }

    public WebElement Button_NoReferralConfirm() { return findElement("android:id/button2", LocatorType.Id); }

    public WebElement Button_NoReferralYes() { return findElement("android:id/button1", LocatorType.Id); }

    public WebElement Snackbar_Error() { return findElement("com.bungii.customer:id/snackbar_text", LocatorType.Id); }

    public WebElement Text_ReferralSourceAdded() { return findElement("com.bungii.customer:id/signup_textview_hear_about_us_choice", LocatorType.Id); }

    //--------------Sign up fields error messages-----------------------------------------------------------
    public WebElement Cust_Signup_Error_Email() { return findElement("//android.widget.TextView[@resource-id='com.bungii.customer:id/textinput_error' and @instance='2']", LocatorType.XPath); }

    public WebElement Cust_Signup_Error_Phone() { return findElement("//android.widget.TextView[@resource-id='com.bungii.customer:id/textinput_error' and @instance='3']", LocatorType.XPath); }

    public WebElement Cust_Signup_Error_Password() { return findElement("//android.widget.TextView[@resource-id='com.bungii.customer:id/textinput_error' and @instance='4']", LocatorType.XPath); }

    //--------------Verification page elements---------------------------------------------------------------

    public WebElement Textfield_SMSCode() { return findElement("com.bungii.customer:id/field_sms_code", LocatorType.Id); }

    public WebElement Button_VerifyContinue() { return findElement("com.bungii.customer:id/smsVerifyContinue", LocatorType.Id); }

    public WebElement Link_Resend() { return findElement("//android.widget.Button[@text='RESEND']", LocatorType.XPath); }

    public WebElement Title_Verification() { return findElement("//android.widget.TextView[@text='VERIFICATION']", LocatorType.XPath); }
}
